lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

```diff
@@ -340,6 +340,7 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
                 $mapping['cascade'] = $oneToOneAnnot->cascade;
                 $mapping['orphanRemoval'] = $oneToOneAnnot->orphanRemoval;
                 $mapping['fetch'] = $this->getFetchMode($className, $oneToOneAnnot->fetch);
+                $mapping['indexName'] = $oneToOneAnnot->indexName;
```

Comment:
I think the less we mess with implicit indexes the better - a better solution would be to define an @Index on the table that satisfies the FK index. Not sure if the TableDiff will prevent creation of the FK index if it's exactly satisfied by an explicit index though.